From f33e91f2db544cd1ae17e1b80612d952e9e42927 Mon Sep 17 00:00:00 2001
From: Kay Lin <kaymw@aosc.io>
Date: Thu, 16 Jun 2022 18:16:29 -0700
Subject: [PATCH 2/2] ntfsck: return on in unsupported filesystem cases

Cherry-picked from Fedora package sources. Prevents unexpected
failures while fscking on boot.

Co-authored-by: Tom Callaway <spot@fedoraproject.org>
---
 ntfsprogs/ntfsck.c |  6 +++++-
 src/lowntfs-3g.c   | 14 --------------
 src/ntfs-3g.c      | 14 --------------
 3 files changed, 5 insertions(+), 29 deletions(-)

diff --git a/ntfsprogs/ntfsck.c b/ntfsprogs/ntfsck.c
index 8c126411..cbd163cc 100644
--- a/ntfsprogs/ntfsck.c
+++ b/ntfsprogs/ntfsck.c
@@ -878,7 +878,11 @@ int main(int argc, char **argv)
 	if (errors)
 		return 2;
 	if (unsupported)
-		return 1;
+		ntfs_log_info("ntfsck was unable to run properly.\n");
+		// If we return 1 here, we fail for ntfs services fscking on boot just because
+		// ntfsck isn't smart enough to handle 99% of cases. So, we just return 0.
+		// return 1;
+		return 0;
 	return 0;
 }
 
diff --git a/src/lowntfs-3g.c b/src/lowntfs-3g.c
index 3300fd2f..34e4dd9a 100644
--- a/src/lowntfs-3g.c
+++ b/src/lowntfs-3g.c
@@ -289,20 +289,6 @@ int restore_privs(void);
 #include "../libfuse-lite/priv.h"
 #endif
 
-static const char *setuid_msg =
-"Mount is denied because setuid and setgid root ntfs-3g is insecure with the\n"
-"external FUSE library. Either remove the setuid/setgid bit from the binary\n"
-"or rebuild NTFS-3G with integrated FUSE support and make it setuid root.\n"
-"Please see more information at\n"
-"https://github.com/tuxera/ntfs-3g/wiki/NTFS-3G-FAQ\n";
-
-static const char *unpriv_fuseblk_msg =
-"Unprivileged user can not mount NTFS block devices using the external FUSE\n"
-"library. Either mount the volume as root, or rebuild NTFS-3G with integrated\n"
-"FUSE support and make it setuid root. Please see more information at\n"
-"https://github.com/tuxera/ntfs-3g/wiki/NTFS-3G-FAQ\n";
-#endif
-
 
 static void ntfs_fuse_update_times(ntfs_inode *ni, ntfs_time_update_flags mask)
 {
diff --git a/src/ntfs-3g.c b/src/ntfs-3g.c
index 1fee91f7..a6a85927 100644
--- a/src/ntfs-3g.c
+++ b/src/ntfs-3g.c
@@ -224,20 +224,6 @@ int restore_privs(void);
 #include "../libfuse-lite/priv.h"
 #endif
 
-static const char *setuid_msg =
-"Mount is denied because setuid and setgid root ntfs-3g is insecure with the\n"
-"external FUSE library. Either remove the setuid/setgid bit from the binary\n"
-"or rebuild NTFS-3G with integrated FUSE support and make it setuid root.\n"
-"Please see more information at\n"
-"https://github.com/tuxera/ntfs-3g/wiki/NTFS-3G-FAQ\n";
-
-static const char *unpriv_fuseblk_msg =
-"Unprivileged user can not mount NTFS block devices using the external FUSE\n"
-"library. Either mount the volume as root, or rebuild NTFS-3G with integrated\n"
-"FUSE support and make it setuid root. Please see more information at\n"
-"https://github.com/tuxera/ntfs-3g/wiki/NTFS-3G-FAQ\n";
-#endif
-
 
 /**
  * ntfs_fuse_is_named_data_stream - check path to be to named data stream
-- 
2.36.0

